build.sh

```diff
@@ -2,8 +2,8 @@
 
 function keep_alive() {
   while true; do
-    date
-    sleep 60
+    echo .
+    read -t 60 < /proc/self/fd/1 > /dev/null 2>&1
```

Comment:
Why were these lines changed ? Can you help me understand what `read -t 60 < /proc/self/fd/1 > /dev/null 2>&1` does ?

Reply:
When available RAM is low, the shell fails to run processes and exits. `date` and `sleep` are new processes. By using `echo` and `read` (from a pipe there's nothing to read from, the `stdout` pipe) with timeout, two shell built-in commands, we can output something and add delay, but without running external processes. This makes CI more reliable.